sdks/go/pkg/beam/transforms/xlang/inference/inference.go

```diff
@@ -0,0 +1,115 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package inference has the cross language implementation of RunInference API implemented in Python SDK.
+// An exapnsion service for python external transforms can be started by running
+//
+//	$ python -m apache_beam.runners.portability.expansion_service_main -p $PORT_FOR_EXPANSION_SERVICE
+package inference
+
+import (
+	"reflect"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/core/runtime/xlangx"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/core/typex"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/transforms/xlang"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/transforms/xlang/python"
+)
+
+func init() {
+	beam.RegisterType(reflect.TypeOf((*sklearnConfig)(nil)).Elem())
+	beam.RegisterType(reflect.TypeOf((*argsStruct)(nil)).Elem())
+	beam.RegisterType(reflect.TypeOf((*sklearnKwargs)(nil)).Elem())
+	beam.RegisterType(reflect.TypeOf((*PredictionResult)(nil)).Elem())
+}
+
+var outputT = reflect.TypeOf((*PredictionResult)(nil)).Elem()
+
+// PredictionResult represents the result of a prediction obtained from Python's RunInference API.
+type PredictionResult struct {
+	Example   []int64 `beam:"example"`
+	Inference int32   `beam:"inference"`
+}
+
+type sklearnConfig struct {
+	kwargs        sklearnKwargs
+	args          argsStruct
+	expansionAddr string
+}
+
+type sklearnConfigOption func(*sklearnConfig)
+
+// WithArgs set arguments for the sklearn inference transform parameters
+func WithArgs(args []string) sklearnConfigOption {
+	return func(c *sklearnConfig) {
+		c.args.args = append(c.args.args, args...)
+	}
+}
+
+// WithExpansionAddr provides URL for Python expansion service.
+func WithExpansionAddr(expansionAddr string) sklearnConfigOption {
```

Comment:
We may need a different tactic here, since WithExpansionAddr is general, but this is attached to a an sklearn specific struct. 

We are either going to need general args handling, that we translate to a specific struct afterwards, or need to do something with interfaces to be able use the same options for each model type call.

Reply:
I agree on this one. I'll think of something but should be doable

Reply:
Done